### ISSUE TITLE
Unable to verify smart contracts: POST /verify_smart_contract/contract_verifications returns NO CONTENT

### ISSUE THREAD

**Issue by isingh108** [blockscout/blockscout#6524]
*Unable to verify smart contracts: POST /verify_smart_contract/contract_verifications returns NO CONTENT*

### Environment

* Deployment type: Docker-compose
* Elixir & Erlang/OTP versions (`elixir -version`): Elixir 1.13.1 & Erlang/OTP 24 [erts-12.2]
* Node JS version (`node -v`):  v16.13.1
* Operating System: ubuntu 20.04 LTS
* Blockscout Version/branchcommit: 4.1.8/master/1b621c724b8a46e8fe0dfcccbb09ed197738c04f
* Archive node type && version (Erigon/Geth/Nethermind/Ganache/?):Polygon-edge

### Steps to reproduce

Set RPC urls in docker-compose/envs/common-blockscout.yml to a **polygon-edge node**. Then, run `docker compose up` from the docker-compose directory. This will start all the services including smart-contract verifier.

### Expected behaviour

*The contract should get verified or it should report an error.*

### Actual behaviour

![Network request screenshot](https://user-images.githubusercontent.com/4157381/204513863-1db6ef16-a3e9-4ec7-abf3-033dcdf30a0c.png)
The contract should be verifable from [this link](http://34.204.180.17:4000/address/0x7Db3F4C5916CCE356Ad97Cfbf90075E8e9e3bc8a/verify-via-flattened-code/new)
Once we click on verify after supplying all the required information. The UI stays in loading state and the network tab displays that the target enpoint returns NO CONTENT.

Inferring from the logs of the smart-contract-verifier service, it seems like the request's contents successfully get forwarded from blockscout.
![Log smart-contract-verifier](https://user-images.githubusercontent.com/4157381/204513740-2efd9f9f-3bec-4b0b-bafe-f05fcfbf851c.png)

Here's the HAR export of the network request.
[34.204.180.17.contract_verifier.zip](https://github.com/blockscout/blockscout/files/10112650/34.204.180.17.contract_verifier.zip)

**Comment by vijay-immanent** [comment]
Hmm 🤔 , I've been browsing the source code to figure out how this works and so far, I've been able to surmise the following:

![Solidity Contract Verifier Controller](https://user-images.githubusercontent.com/101624639/204706438-46b039c1-3834-4453-be01-4e39ad61345a.png)

The relevant controller forwards the contract verification request to a worker and then returns status 204 without waiting for the worker to respond.

![Broadcast verification result](https://user-images.githubusercontent.com/101624639/204706818-0f4dea66-40ce-473a-9e0d-06076015f805.png)

The worker broadcasts the verification results through the Endpoint. I'm still trying to discover the parts which handle this broadcasted message.

**Comment by vijay-immanent** [comment]
I believe the response should be coming back as a websocket message but it doesn't. I kept waiting for more than 10 minutes.
![imgpsh_fullsize_anim (1)](https://user-images.githubusercontent.com/101624639/204729914-f06397d7-9895-4cf2-bc39-47fffe196298.png)

**Comment by nikitosing** [comment]
In your case on this address there is no Code tab (it means that you cannot verify it since you have no indexed bytecode). Mb because on your instance disabled internal txs indexer but this contract was created in internal transaction. Or mb used modified ethereum client is not fully compatible with geth/other client and blockscout cannot fetch bytecode. How did you deploy this contract?
![image](https://user-images.githubusercontent.com/32202610/205435013-d951f461-a910-4e33-b4fa-27797688ff82.png)

**Comment by vijay-immanent** [comment]
@nikitosing you are right. We are using **_polygon-edge_** as our blockchian instead of the popular ethereum clients like geth, parity or nethermind. The chain reports the error "method not available/supported" for many JSON_RPC methods including the method **_eth_getCode_**. 

In fact, blockscout is working perfectly with our other private chain that uses geth.

**Comment by hhq365** [comment]
Hi,@vijay-immanent  Check your blockscout make sure it receiving and processing new blocks normally.Once I had encountered the same issue as you. Then I found my blockscout had suspended, pages could open normal, but have not got new blocks in home page-blocks. I restart my blockscout ,recreate a same contract,go to the contract page, the "code tab" appeared.But I didn't know why it happened.
